Add minimal view

cyril42e opened the pull request:
Hi, I created a new view that I called "minimal", that only displays information about the current window (no other windows, no total, just 2 figures), and preserves a "+" sign in the middle as in the original button, so that it remains clear that it is still the new tab button.

It was mostly based on my own need / preference, but maybe other people would be interested so I tried to do it properly so that it is "mergeable" !

The same idea could by the way be used to highlight (bold, star, ...) the current window for the other views.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a "Minimal View" display mode that shows only the current window's tab counts and a popup option to select it.
* **Documentation**
  * README updated with a new "Minimal View" entry and expanded "Display Mode Examples" (includes a placeholder note for detailed Minimal View examples).
* **Chores**
  * Added a script to streamline packaging the extension.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cyril42e committed:
Add new view "simple" that only shows tabs of the current version

and keeps the "+" display in the middle of the new tab button

cyril42e committed:
Add MakeXPI.sh script for Linux

cyril42e committed:
Optimize simple view by updating only windows that need to be updated (one most of the time)

cyril42e committed:
Rename simple view into minimal view

irvinm commented:
@cyril42e thanks for the pull request.  In general, I like the idea, but have 1 concern and 1 thought.

I like the idea of having a mode that just shows the tab situation in the current window (Maybe we could call it "Current Window" mode vs. "Minimal View" (minor point) but I am not in love with how it looks.

On my machine with Hover:
<img width="323" height="98" alt="image" src="https://github.com/user-attachments/assets/8635ef5a-2203-4d56-8a42-88b83debaded" />

Without Hover:
<img width="320" height="95" alt="image" src="https://github.com/user-attachments/assets/33fd349f-c4aa-4d2d-8887-5ade677de5d4" />

Maybe it makes sense to leave the "carot" and "+" with this mode as there is less information although I would want to fix the vertical alignment across the: carot, loaded tab, "+", total tabs.  I would also want to check on the horizontal alignment as it feels/looks like the "+" isn't centered.

My other thought is that I have been playing around with possibly being able to show BOTH the loaded and total tabs on the addon button in the tool bar.  This would have to be done with dynamic SVG generation only which I have basically working although given the button is "small", this is a lot of information to show in a small space.

Prototype:
<img width="48" height="38" alt="image" src="https://github.com/user-attachments/assets/e9cf2afa-13a2-4cb1-a1b8-8ec81c417f50" />

So ... this provides a large array of supported scenarios between the TST bar and the Firefox addon button.

Off the top of my head the options might go like this:

TST bar:

- Multi-window normal: 1 line per window
- Multi-window compact:  Minimized information of all windows
- Current window / Minimized:  Only current window information

Firefox addon button: (Need to think about how to handle the SVG combinations here too)
- (Normal or SVG rendering) Normal: Total tabs (Normal does handle more than 999 well)
- (SVG Only) Multi-window: Total Loaded / Total Tabs
- (SVG Only) Current window:  Total Loaded Current Window / Total Tabs Current Window

I want to think about how to handle all of this, then look at this PR and incorporate.  (I glanced at the code changes and they looked pretty good, but will inspect more when I look to merge.)

cyril42e committed:
Remove the 15px spacing horizontal

it was put there on purpose to take into account the drop down arrow on hover,
but that's not how it is done natively so do the same

cyril42e committed:
Vertically align the caret

cyril42e committed:
Try to improve centering of + sign

- set size of + sign in em instead of px, for better robustness
- set line-height to 1, to avoid offsets with different font sizes

but the main issue is probably that it depends on how each font is vertically aligning the + on the line

cyril42e committed:
Artificially move a bit the plus sign upwards to compensate for the font offset

not very robust, but lessens a bit the issue

cyril42e commented:
Sure, you can integrate it the way you prefer! I'll just get a developer account to sign and use my version meanwhile :)

About the vertical alignment, I didn't put a lot of care on it because on my system the + symbol is correctly centered,
and the arrow was already not centered with this extension:

<img width="302" height="62" alt="upstream_line_hover" src="https://github.com/user-attachments/assets/795022f0-e414-41db-a311-3583046916ef" />
<img width="302" height="62" alt="branch_minimal_hover" src="https://github.com/user-attachments/assets/c171e18a-5aaa-4a78-bd17-abcdb098d7d3" />

So I had a look at how we can improve it to be more robust and consistent, I fixed it for the caret, and for the plus I made one change that may improve the robustness (line-height=1), but I think that the main issue may be that depending on the font, because the + symbol is not centered compared to digits symbols. So I also added a hard-coded 5% offset for testing, it should improve a bit, but it's not clean... If it is due to the font, the only solution may be to use an icon instead of a character (I didn't find a full-height plus unicode symbol).

About the horizontal alignement, it was on purpose so that it looks centered when hovering (taking into account the arrow). But it is true that natively it is centered in the whole button, so I should have done the same. I have reverted it.

And about the name, let me know if you want me to do the renaming to "Current Window" or if you prefer to do it along with tuning the description strings.
